File: 1684-count-the-number-of-consistent-strings/1684-count-the-number-of-consistent-strings.java
class Solution {
    public int countConsistentStrings(String allowed, String[] words) {
        boolean[] seen = new boolean[26];

        for(int i = 0;i < allowed.length();i++){
            seen[allowed.charAt(i) - 'a'] = true;
        }

        int count = 0;
        for(String word : words){
            boolean valid = true;
            for(int i = 0;i < word.length();i++){
                if(!seen[word.charAt(i) - 'a']){
                    valid = false;
                    break;
                }
            }
            if(valid){
                count++;
            }
        }
        return count;
    }
}

File: 1684-count-the-number-of-consistent-strings/1684-count-the-number-of-consistent-strings.py
class Solution:
    def countConsistentStrings(self, allowed: str, words: List[str]) -> int:
        arr = [False]*26
        for ch in allowed:
            arr[ord(ch) - ord('a')] = True
        
        count = 0
        for word in words:
            valid = True
            for ch in word:
                if not arr[ord(ch) - ord('a')]:
                    valid = False
                    break
            if valid:
                count+=1
        return count

